### ISSUE TITLE
Missing build documentation

### ISSUE THREAD

**Issue by DCtheTall** [DCtheTall/node-video-chat#23]
[Multiple](https://github.com/DCtheTall/node-video-chat/issues/19) [issues](https://github.com/DCtheTall/node-video-chat/issues/22) have brought up that the local build process is confusing or unclear. There should be build instructions added to the README.

**Comment by VictorJL** [comment]
I'm trying to build the server, I managed to compile and run, but I'm at the point of rendering the pug template and it does not work properly.

What are your steps to run the signaling server? Is it bundled with the client? What is the logic behind rendering the React application that is being compiled with Webpack?

**Comment by VictorJL** [comment]
I think the graphql server is the one connected with the React app. 

Heroku gives me an error while deploying the signalServer, but I think it's related to the way we pass it the database url

**Comment by DCtheTall** [comment]
> I think the graphql server is the one connected with the React app.

You are correct! Since we use Apollo's server side rendering algorithm to populate components with some data, the GraphQL server handles serving the views.

The signaling server will just be contacted by the app's internals, once it is up and running you just query the address of the GraphQL server to get the front end. This should probably be documented.

> Heroku gives me an error while deploying the signalServer, but I think it's related to the way we pass it the database url

Can you elaborate? Both servers are given the database URL the same way.

I think the signaling server should probably not even be aware the SQL database exists, which was an oversight on my part.

**Comment by VictorJL** [comment]
This is the [.env](https://gist.github.com/VictorJL/9893eff97c18415756aac3196b90ec8d) file I'm using, it no longer says anything about the database but fails to query, looks like a Postgres version conflict. 

This is one of the errors
```
original: error: operator does not exist: integer = boolean
```
```
[SequelizeDatabaseError]: operator does not exist: integer = boolean
```

What version of Postgres should I be using?

**Comment by VictorJL** [comment]
Inside ContactRequests.js there is another error

```
    try {

      const pendingRequests = await models.contact_request.findAll({
        where: {
          recipient_id: req.user && req.user.id,
          status: models.contact_request.statuses.PENDING,
        },
        include: [{
          model: models.user,
          as: 'sender',
        }],
        order: [['createdAt', 'DESC']],
        limit: 100,
      });
      await Promise.map(pendingRequests, async (request) => {
        if (moment(request.createdAt) < moment().startOf('day').subtract(1, 'month')) {
          request.status = models.contact_request.statuses.EXPIRED;
          await request.save();
        }
      });
      return pendingRequests;
    } catch (err) {
      console.log(err);
      return [];
    }
```

req.user and req.user.id are undefined, which I suppose are passport or jwt validations, for which the error throws:

WHERE parameter "recipient_id" has invalid "undefined" value

**Comment by DCtheTall** [comment]
@VictorJL at this point it's probably better to move this discussion about your errors to another issue. I'll investigate when I have a chance.

**Comment by VictorJL** [comment]
No problem. I'm also up for documenting the setup steps I had to face while running the app.

**Comment by DCtheTall** [comment]
@VictorJL a PR is welcome if you want to add it to the README. I guess my "review" will be trying to reproduce since I haven't manually built this project in quite some time 😆
